File: src/features/profile-settings/domain/repositories/profile-settings-repository.ts
import type { ProfileSettings } from "@/features/profile-settings/domain/entities/profile-settings";

export interface ProfileSettingsRepository {
  /** Load the profile's settings (backend returns lazy defaults if no row). */
  get(profileId: string): Promise<ProfileSettings>;
  /** Persist the full settings row; returns the canonical stored value. */
  set(profileId: string, settings: ProfileSettings): Promise<ProfileSettings>;
}

File: src-tauri/src/domain/repositories/profile_settings_repository.rs
use async_trait::async_trait;

use crate::domain::entities::profile_settings::ProfileSettings;
use crate::domain::error::DomainError;

/// Per-profile settings persistence (over the unlocked vault connection).
/// `load` returns [`ProfileSettings::default`] when the profile has no row
/// (lazy default); `save` upserts the whole row.
#[async_trait]
pub trait ProfileSettingsRepository: Send + Sync {
    async fn load(&self, profile_id: &str) -> Result<ProfileSettings, DomainError>;

    async fn save(&self, profile_id: &str, settings: &ProfileSettings) -> Result<(), DomainError>;
}
